tracing/src/main/java/com/palantir/tracing/DeferredTracer.java

```diff
@@ -34,23 +38,54 @@
  *     return null;
  * });
  *
+ * N.b. the captured trace is restored without the full stack of spans, and so it's not possible to complete spans
+ * not started within the deferred context.
+ *
  * </code>
  * </pre>
  */
-public final class DeferredTracer {
-    private final Optional<Trace> trace;
-    private final Optional<String> operation;
+public final class DeferredTracer implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String UNKNOWN_OPERATION = "deferred";
+
+    private final String traceId;
+    private final boolean isObservable;
+    private final String operation;
+    @Nullable
+    private final String parentSpanId;
 
+    /** Create a new deferred tracer without specifying an operation.
+     * @deprecated use {@link #DeferredTracer(String)}.
+     */
+    @Deprecated
     public DeferredTracer() {
-        this(Optional.empty());
+        this(UNKNOWN_OPERATION);
     }
 
-    public DeferredTracer(String operation) {
-        this(Optional.of(operation));
+    /** Create a new deferred tracer, optionally specifying an operation.
+     * @deprecated use {@link #DeferredTracer(String)}.
+     */
+    @Deprecated
+    public DeferredTracer(Optional<String> operation) {
+        this(operation.orElse(UNKNOWN_OPERATION));
```

Comment:
how about attempting to capture the top span's operation, and use that?

Reply:
done